packages/medusa/src/services/order.ts

```diff
@@ -487,35 +487,43 @@ class OrderService extends TransactionBaseService {
     cartId: string,
     config: FindConfig<Order> = {}
   ): Promise<Order> {
-    const orderRepo = this.activeManager_.withRepository(this.orderRepository_)
+    if (!isDefined(cartId)) {
```

Comment:
suggestion: We could have a `retrieve_` method and then the retrieve, retrieveBy* etc calling that retrieve_ method in order to centralize the code as it is the same except for the retrieval condition. This is something we already do here and there. not necessarily in that pr but just to share my thoughts

Reply:
Yes good idea, I will add it to our backlog.